t/shdict.t

```diff
@@ -999,3 +999,70 @@ number expected, got string
 --- no_error_log
 [alert]
 [crit]
+
+
+
+=== TEST 31: get total used memory
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local ffi = require "ffi"
+            local dogs = ngx.shared.dogs
+            local total_used_bytes = dogs:get_stats()
+            ngx.say("total_used_bytes type: ", type(total_used_bytes))
+        }
+    }
+--- request
+GET /t
+--- response_body
+total_used_bytes type: number
+--- no_error_log
+[error]
+[alert]
+[crit]
+
+
+
+=== TEST 32: get detailed stats
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local ffi = require "ffi"
+            local dogs = ngx.shared.dogs
+            local slot_count = dogs:get_pool_slot_count()
+            local slots = {}
+            for i = 1, slot_count do
+                slots[i] = {}
+            end
+            local total_used_bytes = dogs:get_stats(slots)
```

Comment:
Sorry, I changed my mind again. Seems like it's not worth the complexity and overhead of the API to get all the details in each slot. Indeed, 99% of the times, the user is only interested in the total stats. Sorry again for the inconvenience!

Reply:
Don't worry! No need for apology here. What is important here is providing a API which is both easy to use and performant. I'd like to choose the best balance too.
I agree the user is only interested in the total memory used bytes 99% of times.

Oh, by the way, did you verify the calculation of  the total memory used?
Please make sure the calculation is correct.

I've updated the other patch at https://github.com/openresty/lua-resty-core/compare/master...hnakamur:shdict_get_stats.
Now the difference is this pull request needs dynamic record table allocations for each element of the array table, and the other patch needs dynamic allocations for keys like `i .. '.size'`.
Which do you prefer?

Reply:
By the way, this review is a great learning experience for me to implement an openresty API in the FFI way.
Thanks for your time!